src/Initializer/InputParameters.cpp

```diff
@@ -0,0 +1,462 @@
+    }
+  }
+
+  void warnDeprecatedSingle(const std::string& field) {
+    if (hasField(field)) {
+      visited.emplace(field);
+      logWarning(seissol::MPI::mpi.rank())
+          << "The field" << field
+          << "is no longer in use. You may safely remove it from your parameters file.";
+    }
+  }
+
+  void warnDeprecated(const std::vector<std::string>& fields) {
+    for (const auto& field : fields) {
+      warnDeprecatedSingle(field);
+    }
+  }
+
+  void warnLeftover() {
+    for (const auto& subnodes : node) {
+      auto field = subnodes.first.as<std::string>();
+      if (visited.find(field) == visited.end()) {
+        logWarning(seissol::MPI::mpi.rank()) << "The field" << field << "is not known to SeisSol.";
+      }
+    }
+  }
+
+  void markUnused(const std::string& field) {
+    logDebug(seissol::MPI::mpi.rank())
+        << "The field" << field << "is ignored (regardless of if it exists or not)";
+    visited.emplace(field);
+  }
+
+  ParameterReader subreader(const std::string& subnodeName) {
+    visited.emplace(subnodeName);
+    logDebug(seissol::MPI::mpi.rank()) << "Entering section" << subnodeName;
+    if (hasField(subnodeName)) {
+      return ParameterReader(node[subnodeName], false);
+    } else {
+      logDebug(seissol::MPI::mpi.rank())
+          << "Section" << subnodeName
+          << "not found in the given parameter file. Using an empty reader.";
+      return ParameterReader(node[subnodeName], true);
+    }
+  }
+
+  bool hasField(const std::string& field) { return !empty && node[field]; }
+
+  private:
+  template <typename T>
+  T readUnsafe(const std::string& field) {
+    visited.emplace(field);
+    logDebug(seissol::MPI::mpi.rank()) << "The field" << field << "was read.";
+    try {
+      // booleans are stored as integers
+      if constexpr (std::is_same<T, bool>::value) {
+        return node[field].as<int>() > 0;
+      } else {
+        return node[field].as<T>();
+      }
+    } catch (std::exception& e) {
+      logError() << "Error while reading field" << field << ":" << e.what();
+      return T(); // unreachable. TODO: use compiler hint instead
+    }
+  }
+
+  bool empty;
+  YAML::Node node; // apparently the YAML nodes use a reference semantic. Hence, we do it like this.
+  std::unordered_set<std::string> visited;
+};
+
+void readModel(ParameterReader& baseReader, SeisSolParameters& ssp) {
+  auto reader = baseReader.subreader("equations");
+
+  ssp.model.materialFileName =
+      reader.readOrFail<std::string>("materialfilename", "No material file given.");
+  ssp.model.boundaryFileName = reader.readWithDefault("boundaryfilename", std::string(""));
+
+  ssp.model.gravitationalAcceleration = reader.readWithDefault("gravitationalacceleration", 9.81);
+
+  ssp.model.plasticity = reader.readWithDefault("plasticity", false);
+  ssp.model.tv = reader.readWithDefault("tv", 0.1);
+  ssp.model.useCellHomogenizedMaterial =
+      reader.readWithDefault("usecellhomogenizedmaterial", false);
```

Comment:
default is true
https://github.com/SeisSol/SeisSol/blob/master/src/Reader/readpar.f90#L281